diff --git a/runtime/support.py b/runtime/support.py
index e8ec253..0166dfd 100644
--- a/runtime/support.py
+++ b/runtime/support.py
@@ -322,8 +322,11 @@ def setup(import_roots, zip_safe):
     # Initialize import path
     _initialize_import_path(import_roots, import_prefix)
 
-    # Add hook for package metadata
-    _setup_pkg_resources('pkg_resources')
-    _setup_pkg_resources('pip._vendor.pkg_resources')
+    # Skip adding hooks for package metadata.
+    # Importing pip's vendored pkg_resources on Debian systems adds
+    # Debian-distributed wheels to the front of the module search path.
+    # These then take precedence over modules that are packaged in the
+    # archive, which leads to incorrect behavior when the Debian packages
+    # are the wrong version (many years old), etc.
 
     return True
